src/main/java/it/pagopa/transactions/utils/UUIDUtils.java

```diff
@@ -16,9 +20,14 @@ public String uuidToBase64(UUID uuid) {
         return Base64.encodeBase64URLSafeString(bb.array());
     }
 
-    public UUID uuidFromBase64(String str) {
-        byte[] bytes = Base64.decodeBase64(str);
-        ByteBuffer bb = ByteBuffer.wrap(bytes);
-        return new UUID(bb.getLong(), bb.getLong());
+    public Mono<UUID> uuidFromBase64(String str) {
+        try {
+            byte[] bytes = Base64.decodeBase64(str);
+            ByteBuffer bb = ByteBuffer.wrap(bytes);
+            return Mono.just(new UUID(bb.getLong(), bb.getLong()));
+        } catch (BufferUnderflowException e) {
+            return Mono.error(new InvalidRequestException("Error while decode transactionId"));
+        }
```

Comment:
Given that it's not an asynchronous operation is it right to return `Mono`?

Reply:
You are right, I tried to solve it using an Either let me know what you think 😉